src/kOS/Module/kOSVesselModule.cs

```diff
@@ -0,0 +1,346 @@
+        /// </summary>
+        public void FixedUpdate()
+        {
+            if (initialized)
+            {
+                if (parentVessel.Parts.Count != partCount)
+                {
+                    ClearParts();
+                    HarvestParts();
+                    partCount = parentVessel.Parts.Count;
+                }
+                if (parentVessel.loaded)
+                {
+                    UpdateParameterState();
+                }
+            }
+        }
+
+        /// <summary>
+        /// Update is called once per UI tick.
+        /// </summary>
+        public void Update()
+        {
+        }
+
+        /// <summary>
+        /// LastUpdate is called at the end of the update frame.  Useful if something needs to be
+        /// done after the physics and UI updates have been completed, but before scene or gui rendering.
+        /// </summary>
+        public void LastUpdate()
+        {
+        }
+
+        /// <summary>
+        /// OnGUI is where any vessel specific interface rendering should be done.
+        /// </summary>
+        public void OnGUI()
+        {
+        }
+        #endregion
+
+        /// <summary>
+        /// Generates a list of all processorModules included on the parentVessel.
+        /// </summary>
+        private void HarvestParts()
+        {
+            var proccessorModules = parentVessel.FindPartModulesImplementing<kOSProcessor>();
+            foreach (var proc in proccessorModules)
+            {
+                childParts.Add(proc.part.flightID);
+                uint id = proc.part.flightID;
+                if (partLookup.ContainsKey(id) && partLookup[id].ID != ID)
+                {
+                    // If the part is already known and not associated with this module, then
+                    // it is appearing as a result of a staging or undocking event (or something
+                    // similar).  As such, we need to copy the information from the flight parameters
+                    // on the originating module.
+                    foreach (string key in flightControlParameters.Keys)
+                    {
+                        if (partLookup[id].HasFlightControlParameter(key))
+                        {
+                            // Only copy the data if the previous vessel module still has the parameter.
+                            IFlightControlParameter paramDestination = flightControlParameters[key];
+                            IFlightControlParameter paramOrigin = partLookup[id].GetFlightControlParameter(key);
+                            // We only want to copy the parameters themselves once (because they are vessel dependent
+                            // not part dependent) but we still need to iterate through each part since "control"
+                            // itself is part dependent.
+                            if (partLookup[id].ID != BaseId) paramDestination.CopyFrom(paramOrigin);
```

Comment:
We may want to change this logic to only copy "enabled" parameters, or perhaps only "enabled" parameters who's controlling part is on the new vessel.  This could be done pretty easily by moving the copy line into the if block below.  I'm open to discussion on it though.